guide/api-design-guidelines.md

```diff
@@ -0,0 +1,193 @@
+Please ensure that you add an anchor tag to any new guidelines that you add and maintain the naming convention.
+-->
+
+## History
+
+<details>
+  <summary>Expand change history</summary>
+
+| Date        | Notes                                                          |
+| ----------- | -------------------------------------------------------------- |
+| 2024-Sep-17 | Added guidance on Secrets                                      |
+
+</details>
+
+## Introduction
+
+These are prescriptive guidelines that Radius contributors MUST follow while designing APIs to ensure that customers have a great experience. These guidelines help make Radius APIs:
+- Developer friendly via consistent patterns
+
+Technology and software is constantly changing and evolving, and as such, this is intended to be a living document. [Open an issue](https://github.com/radius-project/design-notes/issues) to suggest a change or propose a new idea. 
+
+### Prescriptive Guidance
+This document offers prescriptive guidance labeled as follows:
+
+:white_check_mark: **DO** adopt this pattern.
+
+:ballot_box_with_check: **YOU SHOULD** adopt this pattern. If not following this advice, you MUST disclose your reason during a design review discussion.
+
+:heavy_check_mark: **YOU MAY** consider this pattern if appropriate to your situation.
+
+:warning: **YOU SHOULD NOT** adopt this pattern. If not following this advice, you MUST disclose your reason during a design review discussion.
+
+:no_entry: **DO NOT** adopt this pattern.
+
+## API Foundation:
+
+Radius provides an HTTP-based API that deploys and manages cloud-native applications as well as on-premise or cloud resources.
+The overall design of the Radius API is based on the Azure Resource Manager API (ARM). Radius generalizes the design of ARM, removes proprietary Azure concepts, and extends the ARM contract to support the conventions of other resource managers like Kubernetes or AWS Cloud-Control. 
+
+Radius utilizes TypeSpec for describing API model definitions and operations, providing a consistent approach to defining the structure of API payloads and interactions. The TypeSpec specification serves as the source to generate corresponding OpenAPI 2.0 (swagger) API documentation.
+
+Reference: 
+- [Radius API](https://docs.radapp.io/concepts/technical/api/)
+- [Radius Architecture](https://docs.radapp.io/concepts/technical/architecture/)
+
+## Guidelines
+
+### Resource Property Design
+
+<a href="#secrets" name="secrets"></a>
+### Secrets
+
+<a href="#secret-store" name="secret-store">:white_check_mark:</a> **DO** use [Applications.Core/secretStores](https://docs.radapp.io/reference/resource-schema/core-schema/secretstore/) resource to store sensitive data, such as passwords, OAuth tokens, and SSH keys. Through the use of this resource, Radius ensures that confidential data is handled in a safe and consistent manner.
+
```

Comment:
Would it be less confusing to call out that there are two different patterns in this section? I'm just thinking about what would be easier for a reader to grok.

Something like:

> The applicable guidance for API design of secrets is slightly different depending on which of these two scenarios is the best fit:
>
> - Storing and retrieving a single (scalar) secret value. eg: An environment variable. 
> - Storing and retrieving a structured (multiple-value) secret. eg: OAuth configuration.